test_scripts/AppIconsFolder/ATF_P_AppIconsFolder_exists_with_ReadWritePermissions.lua

```diff
@@ -0,0 +1,138 @@
+  end)
+  self.mobileSession:ExpectResponse(corIdRAI, { success = true, resultCode = "SUCCESS" })
+end
+
+local function pathToAppFolderFunction(appID)
+  return appWithPermissions .. "/" .. appID .. "_" .. config.deviceMAC .. "/"
+end
+
+local function checkFunction()
+  local status = true
+  local dirExistResult = commonFunctions:Directory_exist(appWithPermissions)
+  if dirExistResult == true then
+    local applicationFileToCheck = appWithPermissions .. "/" .. RAIParameters.appID
+    local applicationFileExistsResult = commonSteps:file_exists(applicationFileToCheck)
+    if applicationFileExistsResult == false then
+      commonFunctions:userPrint(31, RAIParameters.appID .. " icon is not written to folder")
+       status = false
+    end
+    else 
+      commonFunctions:userPrint(31, "IconsFolder folder does not exist in SDL bin folder" )
+      status = false
+    end
+    return status
+end
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_connectMobile()
+  self:connectMobile()
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:Check_SDL_finds_icons_saved_in_AppIconsFolder()
+  self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  self.mobileSession:StartService(7)
+  :Do(function()
+    registerApplication(self)
+    EXPECT_NOTIFICATION("OnHMIStatus", { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE"})
+    :Do(function()
+     local cidPutFile = self.mobileSession:SendRPC("PutFile",
+      {
+        syncFileName = "iconFirstApp.png",
+        fileType = "GRAPHIC_PNG",
+        persistentFile = false,
+        systemFile = false
+      }, "files/icon.png")
+     EXPECT_RESPONSE(cidPutFile, { success = true, resultCode = "SUCCESS" })
+     :Do(function()
+     local cidSetAppIcon = self.mobileSession:SendRPC("SetAppIcon",{ syncFileName = "iconFirstApp.png" })
+     local pathToAppFolder = pathToAppFolderFunction(RAIParameters.appID)
+     EXPECT_HMICALL("UI.SetAppIcon",
+      {
+        syncFileName =
+          {
+            imageType = "DYNAMIC",
+            value = pathToAppFolder .. "iconFirstApp.png"
+          }
+       })
+    :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    EXPECT_RESPONSE(cidSetAppIcon, { resultCode = "SUCCESS", success = true })
+    :ValidIf(function()
+      return checkFunction()
+    end)
+    end)
+    end)
+  end)
+end
+
+--[[ Postconditions ]]
+commonFunctions:newTestCasesGroup("Postconditions")
+function Test.Postcondition_stopSDL()
+  StopSDL()
+end
+
+function Test.Postcondition_deleteCreatedIconsFolder()
+  assert(os.execute( "rm -rf " .. appWithPermissions))
+end
+
+function Test.Postcondition_restoreDefaultValueInIni()
+  commonFunctions:SetValuesInIniFile("AppIconsFolder%s-=%s-.-%s-\n", "AppIconsFolder", 'storage')
```

Comment:
@aarotar, Please replace 'storage' => 'IconsFolder', since this folder is defined at the start of test